File: vincent_transportation_survey_data_cleaning.R
# LOADING DATASET

library(readr)

Undergraduate_Student_Life_Survey_April_23_2026 <- read_csv("Undergraduate_Student_Life_Survey_April 23-2026.csv")

undergrad_survey <- Undergraduate_Student_Life_Survey_April_23_2026

#SUBSETTING 

# Selecting Finished, ResponseId, Q1 (for filtering), and transportation questions Q9-Q16
# Column indices: Finished(6), ResponseId(8), Q1(11), Q9(24), Q10(25), Q10_7_TEXT(26),
#                 Q11(27), Q12(28), Q13(29), Q16(30)

undergrad_survey <- undergrad_survey[c(6, 8, 11, 24:30)]

# Deleting rows where "Finished" = False AND where Q1 (undergraduate student status) = No or NA

library(dplyr)

undergrad_survey <- filter(undergrad_survey, Finished != "False" & Q1 != "No")

# Deleting rows 1 and 2 (question text and import metadata rows — not actual responses)

undergrad_survey <- undergrad_survey[-c(1, 2), ]


################ VARIABLE RECODING ###########################

library(tidyverse)

## Q9 (on-campus vs off-campus residence): On-campus = 1, Off-Campus = 0

undergrad_survey <- undergrad_survey %>%
  mutate(Q9 = case_when(
    Q9 == "On-campus"  ~ 1,
    Q9 == "Off-Campus" ~ 0
  ))


## Q10 (primary mode of transportation to campus)

# Observations assigned a numeric code based on the order they appeared in the survey
# (Personal Car, Carpool, METRO Bus, METRO Rail, Bicycle, Walking, Other)
# verify this order matches the actual survey if re-running

transport_modes <- data.frame(mode = c("Personal Car",
                                       "Carpool",
                                       "METRO Bus",
                                       "METRO Rail",
                                       "Bicycle",
                                       "Walking",
                                       "Other"))

transport_modes$order_id <- seq_len(nrow(transport_modes))

undergrad_survey <- undergrad_survey %>%
  left_join(transport_modes, by = c("Q10" = "mode")) %>%
  mutate(`Q10 ID` = order_id, .after = Q10) %>%
  select(-order_id)


## Q11 (estimated total commute time in minutes): convert from character to numeric

undergrad_survey <- undergrad_survey %>%
  mutate(Q11 = as.numeric(Q11))


## Q12 (number of METRO transfers): convert from character to numeric

undergrad_survey <- undergrad_survey %>%
  mutate(Q12 = as.numeric(Q12))


## Q13 (owns a private vehicle): Yes = 1, No = 0
# Applies only to columns where every observation is "Yes", "No", or NA
# (catches Q13; leaves other columns untouched)

undergrad_survey <- undergrad_survey %>%
  mutate(across(where(~ all(. %in% c("Yes", "No", NA))),
                ~ if_else(. == "Yes", 1, 0)))

## Q16 (how often providing carpooling services): recode using same frequency scale as Q8

frequency_codes <- data.frame(frequency = c("Basically every day",
                                             "A few times a week",
                                             "A few times a month",
                                             "Once a month",
                                             "Less than once a month",
                                             "Once",
                                             "Not at all"))

# assigns numbers 0-6 (0 = "Not at all", 6 = "Basically every day")

frequency_codes$order_id <- rev(seq_len(nrow(frequency_codes)))

library(stringr)

undergrad_survey <- undergrad_survey %>%
  mutate(Q16 = str_squish(as.character(Q16)))

frequency_codes <- frequency_codes %>%
  mutate(frequency = str_squish(as.character(frequency)))

lookup <- setNames(frequency_codes$order_id, frequency_codes$frequency)

undergrad_survey <- undergrad_survey %>%
  mutate(Q16 = lookup[Q16])

# writing the cleaned dataset to a CSV file
# visualizations and analysis will be done using this file

write_csv(undergrad_survey, "cleaned_transportation_survey.csv")

File: vincent_transportation_survey_data_cleaning.py
import pandas as pd

# LOADING DATASET

df = pd.read_csv("Undergraduate_Student_Life_Survey_April 23-2026.csv")

# SUBSETTING
# Keeping all columns from Finished through Q16 (same selection as Kaylee's script)
# Transportation columns (Q9-Q16) will be recoded below; civic engagement columns
# (Q2-Q8) are left as-is for Kaylee's script to handle

df = df[["Finished", "ResponseId", "Q1", "Q2", "Q3", "Q4", "Q5",
         "Q6", "Q7", "Q15", "Q8_1", "Q8_2", "Q8_3", "Q8_4", "Q8_5",
         "Q9", "Q10", "Q10_7_TEXT", "Q11", "Q12", "Q13", "Q16"]]

# Deleting rows 0 and 1 (question text and import metadata — not actual responses)

df = df.drop(index=[0, 1]).reset_index(drop=True)

# Filtering out incomplete responses and non-undergraduates

df = df[(df["Finished"] != "False") & (df["Q1"] != "No")].reset_index(drop=True)


# VARIABLE RECODING (transportation columns only)

## Q9 (on-campus vs off-campus residence): On-campus = 1, Off-Campus = 0

df["Q9"] = df["Q9"].map({"On-campus": 1, "Off-Campus": 0})


## Q10 (primary mode of transportation to campus)
# Assigned a numeric code based on the order options appeared in the survey
# verify this order matches the actual survey if re-running

transport_modes = {
    "Personal Car": 1,
    "Carpool":      2,
    "METRO Bus":    3,
    "METRO Rail":   4,
    "Bicycle":      5,
    "Walking":      6,
    "Other":        7,
}

q10_index = df.columns.get_loc("Q10")
df.insert(q10_index + 1, "Q10 ID", df["Q10"].map(transport_modes))


## Q11 (estimated total commute time in minutes): convert to numeric

df["Q11"] = pd.to_numeric(df["Q11"], errors="coerce")


## Q12 (number of METRO transfers): convert to numeric

df["Q12"] = pd.to_numeric(df["Q12"], errors="coerce")


## Q13 (owns a private vehicle): Yes = 1, No = 0

df["Q13"] = df["Q13"].map({"Yes": 1, "No": 0})


## Q16 (how often providing carpooling services): recode using same frequency scale as Q8
# 0 = "Not at all", 6 = "Basically every day"

frequency_lookup = {
    "Basically every day":    6,
    "A few times a week":     5,
    "A few times a month":    4,
    "Once a month":           3,
    "Less than once a month": 2,
    "Once":                   1,
    "Not at all":             0,
}

df["Q16"] = df["Q16"].str.strip().map(frequency_lookup)


# CREATING CLEAN CSV
# visualizations and analysis will be done using this file

df.to_csv("cleaned_transportation_survey.csv", index=False)

print("Done. cleaned_transportation_survey.csv has been saved.")
